### PR TITLE
Updates button usage docs for clarity

### PR DESCRIPTION
**Pull request by mperrotti**
These changes are not being proposed as completed button usage documentation, just a slight improvement over what was already there.

The content of the page has simply been restructured and given slightly more detail. No new ideas are being presented, but existing documentation (specifically: [CSS component docs](https://primer.style/css/components/buttons) and [Figma Primer Web](https://www.figma.com/file/GCvY3Qv8czRgZgvl1dG6lp/Primer-Web?node-id=2350%3A48)) was referenced to add useful details that may have been missing before.
 
## What to focus on
- Is the document at least as accurate as it was before these changes?
- Is the content easier to navigate and understand than it was before?

## Drawbacks
The document has more sections. I think this makes it easier to navigate, but that might just be my personal preference.

### Before
See live [Button usage docs](https://primer.style/design/ui-patterns/button-usage)

### After
localhost_8000_ui-patterns_button-usage (1)

### CODE REVIEW

File: content/ui-patterns/button-usage.mdx [hunk 1]
```diff
@@ -30,9 +33,66 @@ At GitHub, buttons are a fundamental building block of our products. The followi
...
+
+Most of the time, a button will have an icon and a label for clarity.
+
+![button with label and icon](https://user-images.githubusercontent.com/2313998/129626583-22dce7ca-611f-42c7-9a51-2ba1551bb5f5.png)
+
+#### Icon only
+
+In some cases, an action can be understood without using just an icon. Icon-only buttons can help save space in a cramped area.
+
+Buttons that only use an icon should still have a label to make them accesssible to assistive technology (for example: read aloud by a screenreader)
```
Comment (colebemis) [inline content/ui-patterns/button-usage.mdx:77]: Maybe we should clarify that the label will be invisible to sighted visitors

File: content/ui-patterns/button-usage.mdx [hunk 2]
```diff
@@ -30,9 +33,66 @@ At GitHub, buttons are a fundamental building block of our products. The followi
...
+
+![button with icon but no label](https://user-images.githubusercontent.com/2313998/129626586-378e2b7f-7300-4804-93fb-c88f7f22407e.png)
 
-When designing buttons, keep the text as short but descriptive as possible. Buttons are to be understood by their type and text; if further clarity is required, consider adding an octicon or adding a text description.
+### Counters
+
+Counts can be added as numerical aid in appropriate scenarios by providing valuable context. For example, showing social proof by displaying the number of people watching a repo.
```
Comment (colebemis) [inline content/ui-patterns/button-usage.mdx:83]: This sentence is a bit wordy
```suggestion
Counters can be added in appropriate scenarios to provide context. For example, showing social proof by displaying the number of people watching a repo.
```

File: content/ui-patterns/button-usage.mdx [hunk 3]
```diff
@@ -30,9 +33,66 @@ At GitHub, buttons are a fundamental building block of our products. The followi
...
+
+![button with count inside](https://user-images.githubusercontent.com/2313998/129626580-465111c8-9237-4e33-af42-b2e830ea9cc3.png)
+
+#### Counter attached to a button
+
+A count can be displayed as a separate element that is attached to a button. Only display a count as separate element that is attached to a button when the button uses the "small" size variant and the button type is "Default" or "Outline".
+
+![button with count attached](https://user-images.githubusercontent.com/2313998/129626579-a2f17327-1b6e-43b4-8971-5b9a24f96dda.png)
 
 ### Adaptable full-width
```
Comment (colebemis) [inline content/ui-patterns/button-usage.mdx:97]:
```suggestion
### Full-width
```

**Review by colebemis — APPROVED** [review]
This is a huge improvement ✨ Thanks for doing this, @mperrotti!

Just left a few minor comments.

**Comment by colebemis** [comment]
If it's not too much extra effort, it would be nice to replace the all the screenshots of old buttons with new buttons.

Also, I noticed that we reference the `#design-systems` channel at the bottom of this page. We should either update that to `#primer` or remove the `Support` section entirely.

**Review by vdepizzol — APPROVED** [review]
Looking great, @mperrotti! ✨ ✨ ✨ 

@colebemis I'm down to just merge this quickly and focus on updated images on a different PR 🤗🤗🤗
